apps/console/src/features/saml2-configuration/pages/saml2-configuration.tsx

```diff
@@ -0,0 +1,392 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentityAppsApiException } from "@wso2is/core/exceptions";
+import { hasRequiredScopes } from "@wso2is/core/helpers";
+import { AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { URLUtils } from "@wso2is/core/utils";
+import { Field, Form, FormPropsInterface } from "@wso2is/form";
+import { EmphasizedSegment, PageLayout, PrimaryButton, URLInput } from "@wso2is/react-components";
+import { FormValidation } from "@wso2is/validation";
+import React, { FunctionComponent, MutableRefObject, ReactElement, useEffect, useMemo, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { Dispatch } from "redux";
+import { Divider, Grid, Placeholder, Ref } from "semantic-ui-react";
+import { AppConstants, AppState, FeatureConfigInterface, history } from "../../core";
+import { updateSaml2Configurations, useSaml2Config } from "../api/saml2-configuration";
+import { Saml2ConfigurationConstants } from "../constants/saml2-configuration";
+import {
+    Saml2ConfigAPIResponseInterface,
+    Saml2ConfigFormErrorValidationsInterface,
+    Saml2ConfigFormValuesInterface
+} from "../models/saml2-configuration";
+
+/**
+ * Props for saml2 configuration settings page.
+ */
+type Saml2ConfigurationPageInterface = IdentifiableComponentInterface;
+
+const FORM_ID: string = "saml2-config-form";
+
+/**
+ * saml2 configuration page.
+ */
+export const Saml2ConfigurationPage: FunctionComponent<Saml2ConfigurationPageInterface> = (
+    props: Saml2ConfigurationPageInterface
+): ReactElement => {
+    const { [ "data-componentid" ]: componentId } = props;
+
+    const pageContextRef: MutableRefObject<any> = useRef(null);
+    const formRef: MutableRefObject<FormPropsInterface> = useRef<FormPropsInterface>(null);
+    const url: MutableRefObject<HTMLDivElement> = useRef<HTMLDivElement>();
+
+    const featureConfig : FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+    const allowedScopes : string = useSelector((state: AppState) => state?.auth?.allowedScopes);
```

Comment:
Please check the other places as well :) 

```suggestion
    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
```

Reply:
Fixed with 1d3ef8fd4f615de27060c959ea91a56ca1f3158f